### PR TITLE
Traefik middleware plugin to rewrite OCSP requests

### PR DESCRIPTION
**Pull request by erkiesken**
Issue [SRE-1775](https://starship.atlassian.net/browse/SRE-1775).

The plugin matches a request by its path prefix (i.e. `/ocsp`), extracts the remainder of data from URL path, converts it into binary body contents and rewrites the request from GET to POST with proper headers.

[SRE-1775]: https://starship.atlassian.net/browse/SRE-1775?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+For each plugin, the Traefik static configuration must define the module name (as is usual for Go packages).
+
+The following declaration (given here in YAML) defines a plugin:
+
+```yaml
+# Static configuration
+
+experimental:
+  plugins:
+    oscp:
```
Comment (mpihlak) [inline README.md:36]:
```suggestion
    ocsp:
```

**Review by alekseiplotnikov — APPROVED** [review]
Looks good.

**Comment by mpihlak** [comment]
So how does that actually work - IIRC Traefik doesn't have OCSP support so what would issue those OCSP GET requests? Is the idea that the Nginx (which has OCSP support) would call the Traefik endpoint instead of going for Vault OCSP directly?

**Comment by alekseiplotnikov** [comment]
> So how does that actually work - IIRC Traefik doesn't have OCSP support so what would issue those OCSP GET requests? Is the idea that the Nginx (which has OCSP support) would call the Traefik endpoint instead of going for Vault OCSP directly?

OCSP requests are already routed through Traefik, which sits between Nginx and Vault, acting as an ingress controller to expose the service outside the Kubernetes cluster. The idea is to add a custom Traefik middleware to convert GET requests to POST, without introducing any additional components.

**Review by mpihlak — APPROVED** [review]

**Pull request merged**